lib/EnsEMBL/REST/Model/Phenotype.pm

```diff
@@ -199,4 +200,73 @@ sub fetch_features_by_region {
   return \@phenotype_features;
 }
 
+
+=head fetch_features_by_gene
+
+extract phenotype features by gene identifier or gene symbol
+=cut
+
+sub fetch_features_by_gene {
+  my $self      = shift;
+  my $species   = shift;
+  my $gene      = shift;
+
+  my $c = $self->context();
+
+  my $gene_ad = $c->model('Registry')->get_adaptor($species,'core', 'gene');
+  my $phenfeat_ad = $c->model('Registry')->get_adaptor($species,'variation', 'phenotypefeature');
+  my $slice_ad = $c->model('Registry')->get_adaptor($species, "core", "slice");
+
+  my $include_assoc = $c->request->parameters->{include_associated};
+  my $include_overlap = $c->request->parameters->{include_overlap};
+
+  my $genes = $gene_ad->fetch_all_by_external_name($gene);
+
+  if (scalar @{$genes} == 0){
+    Catalyst::Exception->throw("Gene $gene not found.");
+  }
+
+  my @phenotype_features;
+  while (my $g = shift@{$genes}){
+    my @pfs = @{$phenfeat_ad->fetch_all_by_Gene($g)};
+    if ($include_assoc) {
+      my @pfs_assoc = @{$phenfeat_ad-> fetch_all_by_associated_gene($g->external_name())};
+      push @pfs, @pfs_assoc;
+    }
+    if ($include_overlap){
+      my $gene_specific_slice = $slice_ad->fetch_by_gene_stable_id($g->stable_id());
+      my @pfs_overlap = @{$phenfeat_ad->fetch_all_by_Slice($gene_specific_slice)};
+      push @pfs, @pfs_overlap;
+    }
+    @pfs = uniq @pfs;
+
+    while (my $pf = shift @pfs){
+      my $ontology_accessions = $pf->get_all_ontology_accessions();
+
+      my $record =  { description       => $pf->phenotype_description(),
```

Comment:
I'm not convinced this is precisely the right place for this. The composition of a "phenotype feature" object for serialisation is not necessarily bound to the functioning of our REST API, and could be handled by the PhenotypeFeature itself, or else be placed in lib/EnsEMBL/REST/EnsemblModel . This would help us with future remodelling of our REST service architecture.

Reply:
@nerdstrike - a good point, but this fits with the style of our other endpoints. We want to review all of our data models for the next iteration of REST, so it would be a good time to address it then.

Reply:
As you wish.